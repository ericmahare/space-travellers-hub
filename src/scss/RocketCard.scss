.d-flex {
  display: flex;
}

.rocketCard {
  gap: 20px;
}

.rocketImage {
  width: 220px;
}

.rocketInfo {
  flex-direction: column;
  gap: 10px;
  align-items: flex-start;
}

.rocketTitle {
  font-weight: bold;
  font-size: 18px;
}

.rocketDetails {
  font-size: 16px;
}

button {
  background-color: rgb(75, 75, 247);
  border: none;
  color: white;
  padding: 10px;
  cursor: pointer;
  border-radius: 5px;
}

.cancel-reserve {
  background: white;
  color: rgb(131, 131, 131);
  padding: 8px;
  border-radius: 4px;
  border: 1px solid rgb(7, 7, 7);
  cursor: pointer;
  margin-top: 15px;
}

.mini-btn {
  background: #1083fd;
  color: white;
  border-radius: 4px;
  margin-right: 5px;
  padding: 4px;
}

@media screen and (max-width: 768px) {
  .rocketCard {
    flex-direction: column;
  }
}
