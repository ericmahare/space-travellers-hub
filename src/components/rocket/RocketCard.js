import React from 'react';
import '../../scss/RocketCard.scss'
import { useDispatch } from 'react-redux'
import { reserveRocket } from '../../redux/rocket/rocketSlice';

function RocketCard({ rocket }) {
  const {rocket_id, rocket_description, rocket_name, rocket_img, reserved}  = rocket

  const dispatch = useDispatch();
  const rocketBooking = (id) => {
    dispatch(reserveRocket(id));
  };

  return(
    <section>
      <div className='rocketCard d-flex'>
        <div className='rocketImage'>
          <img src={rocket_img} alt ={rocket_name} className='rocketImage'/>
        </div>

        <div className='rocketInfo d-flex'>
          <span className='rocketTitle'>{rocket_name}</span>
          <p className='rocketDetail'>
            {reserved ? (<span className='mini-btn'>Reserved</span>) : (false)}
            {rocket_description}
          </p>
            <button 
              onClick={() => rocketBooking(rocket_id)}
              type='button'
              className='rocket-btn'
          >
            Reserve Rocket
          </button>
        </div>

      </div>
    </section>
  )
}

export default RocketCard;