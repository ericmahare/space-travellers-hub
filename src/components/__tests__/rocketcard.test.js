import renderer from 'react-test-renderer';
import RocketCard from '../RocketCard';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore'

it('renders correctly', () => {
  const rocket = {
    rocket_id: 1,
    rocket_name: 'Falcon 1',
    rocket_descreption: 'The Falcon 1 was an expendable launch system privaâ€¦launch vehicle to go into orbit around the Eart',
    rocket_img: 'https://imgur.com/DaCfMsj.jpg',
    reserved: false
  }
  const tree = renderer.create(
  <Provider store={store}>
    <RocketCard rocket={rocket}/>
  </Provider>).toJSON()
  expect(tree).toMatchSnapshot();
})