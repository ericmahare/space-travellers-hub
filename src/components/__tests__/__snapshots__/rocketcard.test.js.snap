// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="rocketCard d-flex"
  >
    <div
      className="rocketImage"
    >
      <img
        alt="Falcon 1"
        className="rocketImage"
        src="https://imgur.com/DaCfMsj.jpg"
      />
    </div>
    <div
      className="rocketInfo d-flex"
    >
      <span
        className="rocketTitle"
      >
        Falcon 1
      </span>
      <p
        className="rocketDetail"
      />
      <button
        className="rocket-btn"
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</section>
`;
